shared/actions/kbfs/index.platform.desktop.js

```diff
@@ -42,22 +76,23 @@ function * openInWindows (openPath: string): SagaGenerator<any, any> {
     kbfsPath = yield call(kbfsMountGetCurrentMountDirRpcPromise)
 
     if (!kbfsPath) {
-      console.warn('Error in finding kbfspath')
-      return
+      throw new Error('No kbfsPath')
     }
 
     yield put({type: Constants.changeKBFSPath, payload: {path: kbfsPath}})
   }
 
-  if (kbfsPath) {
-    openPath = path.resolve(kbfsPath, openPath)
-    // Check to make sure our resolved path starts with the kbfsPath
-    // i.e. (not opening a folder outside kbfs)
-    if (!openPath.startsWith(kbfsPath)) {
-      throw new Error(`openInKBFS requires ${kbfsPath} prefix: ${openPath}`)
-    }
-    open(openPath)
+  if (!kbfsPath) {
+    throw new Error('No kbfsPath')
+  }
+
+  openPath = path.resolve(kbfsPath, openPath)
+  // Check to make sure our resolved path starts with the kbfsPath
+  // i.e. (not opening a folder outside kbfs)
+  if (!openPath.startsWith(kbfsPath)) {
+    throw new Error(`openInKBFS requires ${kbfsPath} prefix: ${openPath}`)
   }
+  shell.openItem(openPath)
```

Comment:
Previous code checked whether the path was a directory or a file before opening it. If it was a file, it used `shell.showItemInFolder`. This seems pretty important to make sure we don't open something that will execute. Does that logic happen somewhere else, or should it happen here?